internal/fs/caching_test.go

```diff
@@ -432,3 +436,313 @@ func (t *CachingWithImplicitDirsTest) SymlinksAreTypeCached() {
+	AssertNe(nil, err)
+	AssertThat(err, Error(HasSubstr("no such file or directory")))
+
+	// And we should be able to stat it in bucket1 mount directory.
+	fi, err = os.Stat(path.Join(bucket1MntDir, name))
+	AssertEq(nil, err)
+
+	ExpectEq(name, fi.Name())
+	ExpectEq(len(contents), fi.Size())
+
+	// And read it.
+	b, err := os.ReadFile(path.Join(bucket1MntDir, name))
+	AssertEq(nil, err)
+	ExpectEq(contents, string(b))
+
+	// And overwrite it, and read it back again.
+	err = os.WriteFile(path.Join(bucket1MntDir, name), []byte("burrito"), 0500)
+	AssertEq(nil, err)
+
+	b, err = os.ReadFile(path.Join(bucket1MntDir, name))
+	AssertEq(nil, err)
+	ExpectEq("burrito", string(b))
+
+	AssertEq(nil, os.Remove(path.Join(bucket1MntDir, name)))
+}
+
+func (t *MultiBucketMountCachingTest) FileChangedRemotely() {
+	const name = "foo"
+	const contents = "taco"
+	var fi os.FileInfo
+	var err error
+	bucket1MntDir := getMultiMountBucketDir(bucket1Name)
+
+	// Create an object in GCS.
+	_, err = storageutil.CreateObject(
+		ctx,
+		uncachedBuckets[bucket1Name],
+		name,
+		[]byte(contents))
+
+	AssertEq(nil, err)
+
+	// Create a file via the file system.
+	err = os.WriteFile(path.Join(bucket1MntDir, name), []byte("taco"), 0500)
+	AssertEq(nil, err)
+
+	// Overwrite the object in GCS.
+	_, err = storageutil.CreateObject(
+		ctx,
+		uncachedBuckets[bucket1Name],
+		name,
+		[]byte("burrito"))
+
+	AssertEq(nil, err)
+
+	// Because we are caching, the file should still appear to be the local
+	// version.
+	fi, err = os.Stat(path.Join(bucket1MntDir, name))
+	AssertEq(nil, err)
+	ExpectEq(len("taco"), fi.Size())
+
+	// After the TTL elapses, we should see the new version.
+	cacheClock.AdvanceTime(ttl + time.Millisecond)
+
+	fi, err = os.Stat(path.Join(bucket1MntDir, name))
+	AssertEq(nil, err)
+	ExpectEq(len("burrito"), fi.Size())
+
+	// Reading should work as expected.
+	b, err := os.ReadFile(path.Join(bucket1MntDir, name))
+	AssertEq(nil, err)
+	ExpectEq("burrito", string(b))
+
+	AssertEq(nil, os.Remove(path.Join(bucket1MntDir, name)))
+}
+
+func (t *MultiBucketMountCachingTest) DirectoryRemovedRemotely() {
+	const name = "foo"
+	var fi os.FileInfo
+	var err error
+	bucket1MntDir := getMultiMountBucketDir(bucket1Name)
+
+	// Create a directory via the file system.
+	err = os.Mkdir(path.Join(bucket1MntDir, name), 0700)
+	AssertEq(nil, err)
+
+	// Remove the backing object in GCS.
+	err = uncachedBuckets[bucket1Name].DeleteObject(
+		ctx,
+		&gcs.DeleteObjectRequest{Name: name + "/"})
+
+	AssertEq(nil, err)
+
+	// Because we are caching, the directory should still appear to exist.
+	fi, err = os.Stat(path.Join(bucket1MntDir, name))
+	AssertEq(nil, err)
+	ExpectTrue(fi.IsDir())
+
+	// After the TTL elapses, we should see it disappear.
+	cacheClock.AdvanceTime(ttl + time.Millisecond)
+
+	_, err = os.Stat(path.Join(bucket1MntDir, name))
+	ExpectTrue(os.IsNotExist(err), "err: %v", err)
+
+	// AssertEq(nil, os.RemoveAll(path.Join(bucket1MntDir, name)))
```

Comment:
nit: uncomment or remove this statement.

Reply:
Done. Removed from here and all other places.